var/spack/repos/builtin/packages/slate/package.py

```diff
@@ -25,34 +25,42 @@ class Slate(Package):
     variant('mpi',    default=True, description='Build with MPI support.')
     variant('openmp', default=True, description='Build with OpenMP support.')
 
+    depends_on('bash', type='build')
+    depends_on('scalapack')
+    depends_on('blas')
     depends_on('cuda@9:10', when='+cuda')
-    depends_on('intel-mkl')
-    depends_on('mpi', when='+mpi')
+    depends_on('mpi',       when='+mpi')
 
     conflicts('%gcc@:5')
 
-    def setup_build_environment(self, env):
-        if('+cuda' in self.spec):
-            env.prepend_path('CPATH', self.spec['cuda'].prefix.include)
-        env.prepend_path('CPATH', self.spec['intel-mkl'].prefix.mkl.include)
-
     def install(self, spec, prefix):
-        f_cuda = "1" if spec.variants['cuda'].value else "0"
-        f_mpi = "1" if spec.variants['mpi'].value else "0"
-        f_openmp = "1" if spec.variants['openmp'].value else "0"
+        f_cuda = "1" if spec.variants['cuda'].value   else "0"
+        f_mpi  = "1" if spec.variants['mpi'].value    else "0"
+        f_omp  = "1" if spec.variants['openmp'].value else "0"
 
         comp_cxx = comp_for = ''
         if '+mpi' in spec:
             comp_cxx = 'mpicxx'
             comp_for = 'mpif90'
 
-        make('mpi=' + f_mpi, 'mkl=1', 'cuda=' + f_cuda, 'openmp=' + f_openmp,
-             'CXX=' + comp_cxx, 'FC=' + comp_for)
-        install_tree('lib', prefix.lib)
-        install_tree('test', prefix.test)
-        mkdirp(prefix.include)
-        install('include/slate/slate.hh', prefix.include)
-        install('lapack_api/lapack_slate.hh',
-                prefix.include + "/slate_lapack_api.hh")
-        install('scalapack_api/scalapack_slate.hh',
-                prefix.include + "/slate_scalapack_api.hh")
+        if '^openblas' in spec:
+            blas = 'openblas'
+        elif '^intel-mkl' in spec:
```

Comment:
```suggestion
        elif '^mkl' in spec:
```
Both `intel-mkl` and `intel-parallel-studio+mkl` provide MKL/BLAS libs

Reply:
Thanks for the suggestion, but there is a complication with using intel-parallel-studio because it also provides 'mpi' thus forcing the use of intel's mpi unless you decide to get fancy. I've been unable so far to configure slate using this combination of libraries.